jerry-core/ecma/operations/ecma-array-object.c

```diff
@@ -417,16 +417,15 @@ ecma_op_array_object_define_own_property (ecma_object_t *obj_p, /**< the array o
     }
 
     // e.
-    if (index >= old_len_uint32)
+    if (index < UINT32_MAX
+        && index >= old_len_uint32)
```

Comment:
This seems to be a completely unrelated change. Please have this in a separate patch.